app/models/Resource.java

```diff
@@ -97,64 +82,21 @@ private static String generateId() {
    *          The map to create the resource from
    * @return a Resource containing all given properties
    */
-  @SuppressWarnings("unchecked")
-  public static Resource fromMap(Map<String, Object> aProperties) {
-
-    if (aProperties == null) {
-      return null;
-    }
-
-    String type = (String) aProperties.get(JsonLdConstants.TYPE);
-    String id = (String) aProperties.get(JsonLdConstants.ID);
-    Resource resource = new Resource(type, id);
-
-    for (Map.Entry<String, Object> entry : aProperties.entrySet()) {
-      String key = entry.getKey();
-      Object value = entry.getValue();
-      if (key.equals(JsonLdConstants.ID) && !mIdentifiedTypes.contains(type)) {
-        continue;
-      }
-      if (value instanceof Map<?, ?>) {
-        resource.put(key, Resource.fromMap((Map<String, Object>) value));
-      } else if (value instanceof List<?>) {
-        List<Object> vals = new ArrayList<>();
-        for (Object v : (List<?>) value) {
-          if (v instanceof Map<?, ?>) {
-            vals.add(Resource.fromMap((Map<String, Object>) v));
-          } else {
-            vals.add(v);
-          }
-        }
-        resource.put(key, vals);
-      } else {
-        resource.put(key, value);
-      }
-    }
-
-    return resource;
-
+  public Resource(final Map<String, Object> aProperties) {
+    this((String) aProperties.get(JsonLdConstants.TYPE),
+      (String) aProperties.get(JsonLdConstants.ID));
+    fillFromMap(aProperties, getIdentifiedTypes());
   }
 
-  public static Resource fromJson(JsonNode aJson) {
-    Map<String, Object> resourceMap = new ObjectMapper().convertValue(aJson,
-        new TypeReference<HashMap<String, Object>>() {
-        });
-    return fromMap(resourceMap);
-  }
-
-  public static Resource fromJson(String aJsonString) {
-    try {
-      return fromJson(new ObjectMapper().readTree(aJsonString));
-    } catch (IOException e) {
-      Logger.error("Could not read resource from JSON", e);
-      return null;
-    }
+  public Resource(final JsonNode aJson) {
+    this((Map<String, Object>) OBJECT_MAPPER.convertValue(aJson,
+      new TypeReference<HashMap<String, Object>>() {
+      }));
   }
 
-  public static Resource fromJson(InputStream aInputStream) throws IOException {
-    String json = IOUtils.toString(aInputStream, "UTF-8");
+  public Resource(final InputStream aInputStream) throws IOException {
+    this(IOUtils.toString(aInputStream, "UTF-8"));
     aInputStream.close();
-    return fromJson(json);
   }
 
   public ProcessingReport validate() {
```

Comment:
Shouldn't this, as most other remaining methods, also be moved to `ModelCommon`?

Reply:
No, it's a constructor now.

Reply:
I was referring to the `validate` method: https://github.com/hbz/oerworldmap/blob/1257-interactionsBackend-cleanCommits/app/models/Resource.java#L102

Reply:
Ah, I see. The reason why I did like that it: you can't make access to a static member of a subclass in the abstract (base) class. So, accessing subclass specific `static mSchemaNode` from ModelCommon would be impossible. One could think about making `mSchemaNode` non-static, but a schema node instance for every Resource object appeared quite expensive to mee.